packages/cli/commands/project/cloneApp.js

```diff
@@ -0,0 +1,171 @@
+  uiAccountDescription,
+} = require('../../lib/ui');
+const SpinniesManager = require('../../lib/ui/SpinniesManager');
+const {
+  logApiErrorInstance,
+  ApiErrorContext,
+} = require('../../lib/errorHandlers/apiErrors');
+const { EXIT_CODES } = require('../../lib/enums/exitCodes');
+const { isAppDeveloperAccount } = require('../../lib/accountTypes');
+const {
+  cloneApp,
+  checkCloneStatus,
+  downloadClonedProject,
+} = require('@hubspot/local-dev-lib/api/projects');
+const { getCwd, isValidPath } = require('@hubspot/local-dev-lib/path');
+const { logger } = require('@hubspot/local-dev-lib/logger');
+const { getAccountConfig } = require('@hubspot/local-dev-lib/config');
+const { extractZipArchive } = require('@hubspot/local-dev-lib/archive');
+
+const i18nKey = 'commands.project.subcommands.cloneApp';
+
+exports.command = 'clone-app';
+exports.describe = null; // uiBetaTag(i18n(`${i18nKey}.describe`), false);
+
+exports.handler = async options => {
+  await loadAndValidateOptions(options);
+
+  const accountId = getAccountId(options);
+  const accountConfig = getAccountConfig(accountId);
+  const accountName = uiAccountDescription(accountId);
+
+  trackCommandUsage('clone-app', {}, accountId);
+
+  if (!isAppDeveloperAccount(accountConfig)) {
+    uiLine();
+    logger.error(i18n(`${i18nKey}.errors.invalidAccountTypeTitle`));
+    logger.log(
+      i18n(`${i18nKey}.errors.invalidAccountTypeDescription`, {
+        useCommand: uiCommandReference('hs accounts use'),
+        authCommand: uiCommandReference('hs auth'),
+      })
+    );
+    uiLine();
+    process.exit(EXIT_CODES.ERROR);
+  }
+
+  const { appId } =
+    'appId' in options
+      ? options
+      : await selectPublicAppPrompt({
+          accountId,
+          accountName,
+          options,
+          migrateApp: false,
+        });
+
+  const publicApps = await fetchPublicAppOptions(accountId, accountName);
+  const selectedApp = publicApps.find(a => a.id === appId);
+  if (!selectedApp) {
+    logger.error(i18n(`${i18nKey}.errors.invalidAppId`, { appId }));
+    process.exit(EXIT_CODES.ERROR);
+  }
+
+  const { location } =
+    'location' in options
+      ? options
+      : await promptUser({
+          name: 'location',
+          message: i18n(`${i18nKey}.enterLocation`),
+          default: path.resolve(getCwd(), selectedApp.name),
+          validate: input => {
+            if (!input) {
+              return i18n(`${i18nKey}.errors.locationRequired`);
+            }
+            if (fs.existsSync(input)) {
+              return i18n(`${i18nKey}.errors.invalidLocation`);
+            }
+            if (!isValidPath(input)) {
+              return i18n(`${i18nKey}.errors.invalidCharacters`);
+            }
+            return true;
+          },
+        });
+
+  try {
+    SpinniesManager.init();
+
+    SpinniesManager.add('cloneApp', {
+      text: i18n(`${i18nKey}.cloneStatus.inProgress`),
+    });
+
+    const cloneResponse = await cloneApp(accountId, appId);
```

Comment:
You could destructure the responses from cloneApp and poll inline to shorten this section